bids/layout/index.py

```diff
@@ -223,7 +226,11 @@ def _index_dir(self, path, config, force=None):
                 root=self._layout._root,
             )
             if force is not False:
-                self._index_dir(d, config, force=force)
+                dir_fo = self._index_dir(d, config, force=force)
+                if dir_fo:
+                    all_file_objs += dir_fo
```

Comment:
This is accounting for L188 returning `None`. Now that it returns a list, have it return an empty list there.

```suggestion
                all_file_objs.extend(self._index_dir(d, config, force=force))
```